CANTroller2/src/cantroller2.cpp

```diff
@@ -334,8 +325,9 @@ void loop() {
     // if (boot_button) printf ("hrz: %ld | saf: %ld | out %ld", ctrl_pos_adc[HORZ][FILT], steer_pulse_safe_us, steer_pulse_out_us);
 
     // Voltage of vehicle battery
+    // NOTE: we have these same lines of code above, are they both needed?
     ignition_sense = read_battery_ignition();  // Updates battery voltage reading and returns ignition status
-    if (sim_battery && pot_overload == battery) battery_filt_v = map (pot_filt_percent, 0.0, 100.0, 0.0, battery_max_v);
+    if (simulator.can_simulate(SimOption::battery) && simulator.get_pot_overload() == SimOption::battery) battery_filt_v = map (pot.get_filtered_value(), 0.0, 100.0, 0.0, battery_max_v);
```

Comment:
This conditional then map code is repeated like 6 times so far, could we refactor it? Like a function or class method for a sensor could be used like temps_f[ENGINE] = sensor.read(SimOption::coolant, pot.get_filtered_value());
```
float read(SimOption option, float potValue) {
    if (canSimulate(option)) {
      return simulate(option, potValue);
    }   
    return readAndFilter();
  }
```

Reply:
Another idea is to pass this new Simulator class to the Sensors like 
```
class Sensor {
  Sensor(Simulator* simulator) {
    this->simulator = simulator; 
  }

```
Or to have it as a member like 
```
class Sensor {
  Simulator simulator;

  float read() {
    if (simulator.shouldSimulate()) {
```

Reply:
This is already handled in Sensors via the ControllerMode logic. All of this goop is just to work with the old boolean simulation logic. As we convert components to devices, all this will go away (again, see PressureSensor).